### ISSUE TITLE
Bug: Recipe not following default consume rule

### ISSUE THREAD

**Issue by sniporbob** [grocy/grocy#1979]
In some cases/configurations, a recipe will not respect _any_ of the consume rules.

Example private instance: https://test-1onvlp7yay26m3q7ibwuf3.demo-prerelease.grocy.info/recipes?recipe=7
Offending recipe: Hot dog - meal
Offending ingredient: Hot dog buns

Setup: A parent inventory item of Hot Dog Bun was created. Two children, a Ballpark brand and a Natures Own brand, were created as children. An inventory update was first performed on the Natures Own, setting qty of buns to 8. Ballpark was set via inventory next, also to a qty of 8. Natures Own was set with an earlier due date than Ballpark.

In summary, Natures Own was input first, and has a more recent due date, than Ballpark. Optionally, Natures Own can be marked as opened without changing the result.

In the recipe, the parent category of Hot Dog Bun was used.

Result: Ballpark buns are utilized in the recipe, but Natures Own is the one that should be used instead.
